servatrice/src/serversocketinterface.cpp

```diff
@@ -1000,3 +1002,108 @@ Response::ResponseCode ServerSocketInterface::cmdReloadConfig(const Command_Relo
     settingsCache->sync();
     return Response::RespOk;
 }
+
+Response::ResponseCode ServerSocketInterface::cmdAdjustMod(const Command_AdjustMod &cmd, ResponseContainer & /*rc*/) {
+
+    QString adminName = QString::fromStdString(cmd.admin_name());
+    QString userName = QString::fromStdString(cmd.user_name());
+    QList<ServerSocketInterface *> userList;    //list is used to allow functionality of sending event messages to multiple people if need be.
+
+    if (cmd.should_be_mod()) {
+        logDebugMessage("Received admin command: promote user");
+        QSqlQuery *query = sqlInterface->prepareQuery(
+                "update {prefix}_users set admin = :adminlevel where name = :username");
+        query->bindValue(":adminlevel", "2");
+        query->bindValue(":username", userName);
+        if (!sqlInterface->execSqlQuery(query)) {
+            ServerSocketInterface *admin = static_cast<ServerSocketInterface *>(server->getUsers().value(adminName));
+            userList.append(admin);
+            Event_NotifyUser sendtoadmin;
+            sendtoadmin.set_type(Event_NotifyUser::PROMOTIONFAIL);
+            for (int i = 0; i < userList.size(); ++i) {
+                SessionEvent *se = userList[i]->prepareSessionEvent(sendtoadmin);
+                userList[i]->sendProtocolItem(*se);
+                delete se;
+            }
+            return Response::RespInternalError;
+        }
+
+        ServerSocketInterface *user = static_cast<ServerSocketInterface *>(server->getUsers().value(userName));
+        userList.append(user);
+
+        Event_NotifyUser sendtouser;
+        sendtouser.set_type(Event_NotifyUser::PROMOTED);
+        if (!userList.isEmpty()) {
+            for (int i = 0; i < userList.size(); ++i) {
+                SessionEvent *se = userList[i]->prepareSessionEvent(sendtouser);
+                userList[i]->sendProtocolItem(*se);
+                delete se;
+            }
+        }
+        userList.clear();
+
+        ServerSocketInterface *admin = static_cast<ServerSocketInterface *>(server->getUsers().value(adminName));
+        userList.append(admin);
+
+        Event_NotifyUser sendtoadmin;
+        sendtoadmin.set_type(Event_NotifyUser::PROMOTIONSUCCESS);
+        if (!userList.isEmpty()) {
```

Comment:
The `if` here is redundant because an empty `userList` will result in a `for` check of `0 < 0` and it will never enter the body